packages/unlock-js/src/PublicLock/v12/withdrawFromLock.js

```diff
@@ -0,0 +1,53 @@
+import utils from '../../utils'
+
+/**
+ * Triggers a transaction to withdraw funds from the lock and assign them to the owner.
+ * By default (amount=0), this withdraws all funds.
+ * @param {PropTypes.address} lockAddress
+ * @param {string} beneficiary the address that will receive the withdrawn tokens
+ * @param {string} tokenAddress the token address to withdraw from (zero address is ETH)
+ * @param {string} amount
+ * @param {number} decimals
+ * TODO: get the decimal from the ERC20 contract
+ * @param {function} callback invoked with the transaction hash
+ */
+export default async function (
+  { lockAddress, amount = '0', decimals = 18 },
+  transactionOptions = {},
+  callback
+) {
+  const lockContract = await this.getLockContract(lockAddress)
+  const tokenAddress = await lockContract.tokenAddress()
+
+  // use the signer as beneficiary
+  const beneficiary = await this.signer.getAddress()
```

Comment:
Can we also add the beneficiary as an option in the params and only use the signer if none is set?